sys/conf/ldscript.mips.cheri

```diff
@@ -0,0 +1,319 @@
+  .rel.got       : { *(.rel.got)		}
+  .rela.got      : { *(.rela.got)		}
+  .rel.sdata     :
+    {
+      *(.rel.sdata)
+      *(.rel.sdata.*)
+      *(.rel.gnu.linkonce.s.*)
+    }
+  .rela.sdata     :
+    {
+      *(.rela.sdata)
+      *(.rela.sdata.*)
+      *(.rela.gnu.linkonce.s.*)
+    }
+  .rel.sbss      :
+    { 
+      *(.rel.sbss)
+      *(.rel.sbss.*)
+      *(.rel.gnu.linkonce.sb.*)
+    }
+  .rela.sbss     :
+    {
+      *(.rela.sbss)
+      *(.rela.sbss.*)
+      *(.rel.gnu.linkonce.sb.*)
+    }
+  .rel.sdata2    :
+    {
+      *(.rel.sdata2)
+      *(.rel.sdata2.*)
+      *(.rel.gnu.linkonce.s2.*)
+    }
+  .rela.sdata2   :
+    {
+      *(.rela.sdata2)
+      *(.rela.sdata2.*)
+      *(.rela.gnu.linkonce.s2.*)
+    }
+  .rel.sbss2     :
+    {
+      *(.rel.sbss2)
+      *(.rel.sbss2.*)
+      *(.rel.gnu.linkonce.sb2.*)
+    }
+  .rela.sbss2    :
+    {
+      *(.rela.sbss2)
+      *(.rela.sbss2.*)
+      *(.rela.gnu.linkonce.sb2.*)
+    }
+  .rel.bss       :
+    {
+      *(.rel.bss)
+      *(.rel.bss.*)
+      *(.rel.gnu.linkonce.b.*)
+    }
+  .rela.bss      :
+    {
+      *(.rela.bss)
+      *(.rela.bss.*)
+      *(.rela.gnu.linkonce.b.*)
+    }
+  .rel.plt       : { *(.rel.plt)		}
+  .rela.plt      : { *(.rela.plt)		}
+
+  PROVIDE (_rel_dyn = ALIGN(0x8));
+  .rel.dyn	 : { *(.rel.dyn)		}
+  PROVIDE (_erel_dyn = .);
+
+  .rela.dyn	 : { *(.rela.dyn)		}
+  __cap_relocs : {
+    *(__cap_relocs)
+  }
+  .init          :
+  {
+    KEEP (*(.init))
+  } =0x1000000
+  .reginfo : { *(.reginfo) }
+  .sdata2   : { *(.sdata2) *(.sdata2.*) *(.gnu.linkonce.s2.*) }
+  .sbss2   : { *(.sbss2) *(.sbss2.*) *(.gnu.linkonce.sb2.*) }
+  . = ALIGN(0x2000) + (. & (0x2000 - 1));
+  .data    :
+  {
+    *(.data)
+    *(.data.*)
+    *(.gnu.linkonce.d.*)
+  }
+  .data1   : { *(.data1) }
+    .eh_frame : { KEEP (*(.eh_frame)) }
+  .gcc_except_table : { *(.gcc_except_table) }
+  .ctors   :
+  {
+    /* gcc uses crtbegin.o to find the start of
+       the constructors, so we make sure it is
+       first.  Because this is a wildcard, it
+       doesn't matter if the user does not
+       actually link against crtbegin.o; the
+       linker won't look for a file to match a
+       wildcard.  The wildcard also means that it
+       doesn't matter which directory crtbegin.o
+       is in.  */
+    KEEP (*crtbegin.o(.ctors))
+    /* We don't want to include the .ctor section from
+       from the crtend.o file until after the sorted ctors.
+       The .ctor section from the crtend file contains the
+       end of ctors marker and it must be last */
+    KEEP (*(EXCLUDE_FILE (*crtend.o ) .ctors))
+    KEEP (*(SORT(.ctors.*)))
+    KEEP (*(.ctors))
+  }
+   .dtors         :
+  {
+    KEEP (*crtbegin.o(.dtors))
+    KEEP (*(EXCLUDE_FILE (*crtend.o ) .dtors))
+    KEEP (*(SORT(.dtors.*)))
+    KEEP (*(.dtors))
+  }
+  .plt      : { *(.plt)	}
+  _gp = ALIGN(16) + 0x7ff0;
+  .got		  : { *(.got.plt) *(.got) }
+  _captable = ALIGN(0x20);
```

Comment:
LLD should already be defining `__cap_table_start` and `__cap_table_end`. Can those be used instead? There is also `_CHERI_CAPABILITY_TABLE_` but there's no end symbol for that.

Is there anything else that's different from the base linker script?
The
```
  __cap_relocs : {
    *(__cap_relocs)
  }
```
section should be fine to add to the MIPS linker script since it won't do anything if the section doesn't exist.

Reply:
The linker script is from a time when we did not have as much stability, probably can be ditched now. I'm finishing a merge and then go back to testing that.

Reply:
There are a bunch of extra ALIGN directives that have been added as a result of fb7f00b8c4aea68ceaf041f2340494b53fab6645
Some of the start/end symbols have been defined e5e258346f4edc1926223fef9f776fc723b39ea8 for capability relocations.
There is also some explicit padding that I believe was needed for the FPGA image ef3bc0a1cb41feec995739af7c4511cfda33fff5. If we are comfortable moving everything to the normal MIPS ldscript I can merge them. Otherwise I can just change the start/end section names I'm using.

Reply:
I'd be happy to have it merged to the normal MIPS ldscript. If people feel that this is not okay, I'd vote for adding a ldscript.mips.template with #ifdefs that we turn into the appropriate one with cpp.